pilot/pkg/proxy/envoy/v2/lds.go

```diff
@@ -24,15 +24,19 @@ import (
 	"istio.io/istio/pilot/pkg/model"
 )
 
-func (s *DiscoveryServer) pushLds(con *XdsConnection, push *model.PushStatus, onConnect bool) error {
+func (s *DiscoveryServer) pushLds(con *XdsConnection, push *model.PushStatus, onConnect bool, version string) error {
 	// TODO: Modify interface to take services, and config instead of making library query registry
 
 	rawListeners, err := s.generateRawListeners(con, push)
 	if err != nil {
 		return err
 	}
 	con.HTTPListeners = rawListeners
-	response := ldsDiscoveryResponse(rawListeners, *con.modelNode)
+	response := ldsDiscoveryResponse(rawListeners, *con.modelNode, version)
+	if version != versionInfo() {
+		// Just report for now - after debugging we can suppress the push.
+		adsLog.Warnf("LDS: overlap %s %s %s", con.ConID, version, versionInfo())
```

Comment:
how is this overlap?

Reply:
Change1 -> push1
Change2 (after few seconds ) -> push2

push1 may take 10 seconds and be slower - and a sidecar may get  LDS from push2 first, followed by push1 - which will be out of date.

I haven't seen this in logs (for the time I've tested this PR) - but it can happen. 

The solution is to detect that push2 has started - since the version passed in the param will be different from the global version - and stop and push (and computation ) in push1, saving some memory/CPU/pushes in the process. But this PR is only adding the param and the log for LDS.

Reply:
why can't pushes be serialized to start with? halting a push half way means envoy will have partial info for that push and might freak out before the next push info arrives

Reply:
Also, I meant "overlap" here is a term only you and I would understand.. Users wont. So, you might want to phrase this differently